messaging/writer.go

```diff
@@ -73,6 +74,17 @@ func (mw *missyWriter) Write(key []byte, value []byte) error {
 	return mw.brokerWriter.WriteMessages(context.Background(), msg)
 }
 
+// Write new message with retry
+func (mw *missyWriter) WriteWithRetryCounter(key []byte, value []byte, retryCount int) error {
+	msg := Message{
+		Key:   key,
+		Value: value,
+		RetryCounter: retryCount,
+	}
+	return mw.brokerWriter.WriteMessages(context.Background(), msg)
```

Comment:
ok so actually if you take a look into WriteMessage impl. It uses just key and value and pass that into underlaying kafka implementation (kafka-go). So we are not using the retryCount anyway. To get this working we should probably wrap value bytes as internal structure that has original bytes data and additional one (like retries).